### PR TITLE
[SQ MERGED] Configurable max data points 

### PR DESCRIPTION
**Pull request by yashsarma**
cc @paulnivin 

screen shot 2016-01-03 at 2 36 02 pm

### CODE REVIEW

**Comment by paulnivin** [comment]
:+1:

File: expr.go [hunk 1]
```diff
@@ -960,21 +960,31 @@ func evalExpr(e *expr, from, until int32, values map[metricRequest][]*metricData
...
+		if failure_threshold > max_data_points {
+			logger.Logf("threshold must be lesser than max data points: %d > %d\n", failure_threshold, max_data_points)
+			return nil
+		}
+
 		var results []*metricData
 		for _, a := range args {
 			r := *a
-			r.Name = proto.String(fmt.Sprintf("%s threshold:%d", a.GetName(), threshold))
+			r.Name = proto.String(fmt.Sprintf("%s threshold: %d maxDataPoints: %d", a.GetName(), failure_threshold, max_data_points))
```
Comment (zackhsi) [inline expr.go]: Would be clearer to display
`<alias> threshold: 9 of 10`

**Comment by zackhsi** [comment]
+1 after comment

**Comment by yashsarma** [comment]
:rocket:

**Comment by lyft-buildnotify** [comment]
Submit queue bot says: Your commit has been enqueued for commit to master followed by a test run. I will close and update this PR once tests have passed and your code is ready for deployment...

**Comment by lyft-buildnotify** [comment]
Commit: https://github.com/lyft/carbonapi/commit/fccd98b7ab1c095a1879890424a8faac3cff83f2
Job URL: https://jenkins.lyft.net/job/submit_queue_rebase/10991/

_Your PR has been rebased and pushed to master. This PR has been closed and is referenced by the master commit. The branch has been deleted._

If you need to revert your commit, type '_:sob:_' in a comment.

If you need to make a hotfix on top of the latest deployed revision, type '_submit this please to hotfix branch_' in a comment.

If you need to cherry-pick this into a release branch, type '_:cherries: into `branch_name`_' in a comment

Need more help? Look [here](https://github.com/lyft/ops/blob/master/buildnotify/Submit_Queue.md).

**Pull request closed without merging**